homeassistant/components/input_box.py

```diff
@@ -0,0 +1,265 @@
+    return cfg
+
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        cv.slug: vol.All({
+            vol.Optional(CONF_NAME): cv.string,
+            vol.Optional(CONF_MIN, default=0): vol.Coerce(float),
+            vol.Optional(CONF_MAX, default=100): vol.Coerce(float),
+            vol.Optional(CONF_INITIAL): cv.string,
+            vol.Optional(CONF_STEP, default=1):
+                vol.All(vol.Coerce(float), vol.Range(min=1e-3)),
+            vol.Optional(CONF_ICON): cv.icon,
+            vol.Optional(ATTR_UNIT_OF_MEASUREMENT): cv.string,
+            vol.Optional(CONF_MODE, default=MODE_STR): vol.In(VALID_MODES),
+            vol.Optional(CONF_DISABLED, default=False): cv.boolean,
+        }, _cv_input_box)
+    })
+}, required=True, extra=vol.ALLOW_EXTRA)
+
+
+@bind_hass
+def select_value(hass, entity_id, value):
+    """Set input_box to value."""
+    hass.services.call(DOMAIN, SERVICE_SELECT_VALUE, {
+        ATTR_ENTITY_ID: entity_id,
+        ATTR_VALUE: value,
+    })
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Set up an input box."""
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+
+    entities = []
+
+    for object_id, cfg in config[DOMAIN].items():
+        name = cfg.get(CONF_NAME)
+        minimum = cfg.get(CONF_MIN)
+        maximum = cfg.get(CONF_MAX)
+        initial = cfg.get(CONF_INITIAL)
+        step = cfg.get(CONF_STEP)
+        icon = cfg.get(CONF_ICON)
+        unit = cfg.get(ATTR_UNIT_OF_MEASUREMENT)
+        mode = cfg.get(CONF_MODE)
+        disabled = cfg.get(CONF_DISABLED)
+
+        entities.append(InputBox(
+            object_id, name, initial, minimum, maximum, step, icon, unit,
+            mode, disabled))
+
+    if not entities:
+        return False
+
+    @asyncio.coroutine
+    def async_select_value_service(call):
+        """Handle a calls to the input box services."""
+        target_inputs = component.async_extract_from_service(call)
+
+        tasks = [input_box.async_select_value(call.data[ATTR_VALUE])
+                 for input_box in target_inputs]
+        if tasks:
+            yield from asyncio.wait(tasks, loop=hass.loop)
+
+    hass.services.async_register(
+        DOMAIN, SERVICE_SELECT_VALUE, async_select_value_service,
+        schema=SERVICE_SELECT_VALUE_SCHEMA)
+
+    yield from component.async_add_entities(entities)
+    return True
+
+
+class InputBox(Entity):
+    """Represent a box."""
+
+    def __init__(self, object_id, name, initial, minimum, maximum, step, icon,
+                 unit, mode, disabled):
+        """Initialize a select input."""
+        self.entity_id = ENTITY_ID_FORMAT.format(object_id)
+        self._name = name
+        self._current_value = initial
+        self._minimum = minimum
+        self._maximum = maximum
+        self._step = step
+        self._icon = icon
+        self._unit = unit
+        self._mode = mode
+        if self._mode == MODE_FLOAT:
+            self._pattern = '[\-\.0-9]*'
```

Comment:
Why would we include those patterns in Python? We can just include them in the frontend only.